### PR TITLE
Add entry point routes "/"

### PR DESCRIPTION
**Pull request by martinklepsch**
card: https://trello.com/c/wQXEbxey

See comments in the entry namespace for some potential discussion points.

### CODE REVIEW

**Comment by martinklepsch** [comment]
A couple of lines ended up pretty long let me know if that should be fixed.

**Comment by belucid** [comment]
Reviewing.

File: src/open_company/resources/company.clj [hunk 1]
```diff
@@ -232,6 +232,14 @@
   (vec (sort-by primary-key
     (common/read-resources table-name [primary-key "name"]))))
 
+(defn get-companies-by-index
```
Comment (belucid) [inline src/open_company/resources/company.clj]: this is perfect.

File: src/open_company/api/entry.clj [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            [open-company.resources.company :as company-res]
+            [open-company.representations.common :as common-rep]
+            [open-company.representations.company :as company-rep]
+            [cheshire.core :as json]))
+
+;; ----- Representations
+;; I assume these could go into a separate representations ns but since
+;; they're just links and not full domain objects I just kept them here
+
+(defn company-link [{:keys [name] :as c}]
```
Comment (belucid) [inline src/open_company/api/entry.clj]: I think I'd rather have them in the representation ns. I agree it doesn't warrant it's own namespace, but let's put them in representations.common

**Comment by belucid** [comment]
FWIW @martinklepsch I think the 80 column rule is pretty antiquated personally (the last time I used an 80 column displays was on my Commodore 128) and often results in harder to read code as the line break obscures the code more than the length of the line does.

I personally set my end of line ruler in my editor at 120 columns and try to keep lines under 120... but even that I don't sweat too much if there is a good reason to go over (often an end of line comment).

We have our bikeshed code checker set to complain about lines over 120 with this alias:

```
    "bikeshed!" ["bikeshed" "-v" "-m" "120"] ; code check with max line length warning of 120 characters
```

File: src/open_company/api/entry.clj [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+(defresource entry-point []
+  common/anonymous-resource
+
+  :allowed-methods [:options :get]
+  ;; slug supplied to allow-anonymous here is irrelevant.
+  ;; the allowed? checking and token decoding should be
+  ;; decoupled and probably use :initialize-context for
+  ;; token decoding
+  :allowed? (fn [ctx] (common/allow-anonymous "foo" ctx))
```
Comment (belucid) [inline src/open_company/api/entry.clj]: Hmm.... this seems a little too dirty to me. Let's untangle this and clean it up so we don't have to pass a fake slug here.

File: src/open_company/api/entry.clj [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  :available-media-types ["application/json"]
+
+  :handle-not-acceptable (fn [_] (common/only-accept 406 "application/json"))
+  :handle-unsupported-media-type (fn [_] (common/only-accept 415 "application/json"))
+
+  :handle-ok (fn [{:keys [user] :as _ctx}]
```
Comment (belucid) [inline src/open_company/api/entry.clj]: It borders on being just a personal preference, but I like to keep these Liberator methods in the resource macro down to just 1 liners. I'd prefer this function to be named and moved out of here.

**Comment by belucid** [comment]
@martinklepsch normally you'd write more in the PR description about how you tested things so the PR reviewer could repeat (and hopefully improve on) your test steps. See other closed PR's in the repos for examples if you're not sure what kind of things to include.

File: src/open_company/api/entry.clj (hunk 2 above)
Comment (martinklepsch) [inline src/open_company/api/entry.clj]: I think I agree that `company-link` could go into `common` but the other one `links` doesn't seem to be particularly `common` or "shared" so not sure I agree about moving that as well. 

By moving we introduce another level of indirection in a place where the indirection isn't required or useful in any way. My 2 cents.

File: src/open_company/api/entry.clj (hunk 3 above)
Comment (martinklepsch) [inline src/open_company/api/entry.clj]: Totally agree. I thought this would better be done in another PR after this is merged as it affects all API namespaces and would make things slightly messy to review if done in this one as well.

File: src/open_company/api/entry.clj (hunk 4 above)
Comment (martinklepsch) [inline src/open_company/api/entry.clj]: done https://github.com/open-company/open-company-api/commit/bce1859c99e2ba4d3655b5d95e50dc821e970bc5
